src/result-helpers.ts

```diff
@@ -41,3 +45,49 @@ export const normalize = <A extends unknown[]>(
     }),
   );
 };
+
+export const logServerError = (
+  error: HttpError,
+  {
+    logger,
+    request,
+    input,
+  }: {
+    logger: ActualLogger;
+    request: Request;
+    input: FlatObject | null;
+  },
+) =>
+  !error.expose &&
+  logger.error("Server side error", {
+    error,
+    url: request.url,
+    payload: input,
+  });
+
+/**
+ * @deprecated use ensureHttpError().statusCode instead
+ * @todo remove in v21
+ * */
+export const getStatusCodeFromError = (error: Error): number => {
+  if (isHttpError(error)) return error.statusCode;
+  return error instanceof InputValidationError ? 400 : 500;
+};
+
+/**
+ * @example InputValidationError —> BadRequest(400)
+ * @example Error —> InternalServerError(500)
+ * */
+export const ensureHttpError = (error: Error): HttpError =>
+  isHttpError(error)
+    ? error
+    : createHttpError(
+        error instanceof InputValidationError ? 400 : 500,
+        getMessageFromError(error),
+        { cause: error.cause || error },
+      );
+
+export const getPublicErrorMessage = (error: HttpError): string =>
+  process.env.NODE_ENV === "production" && !error.expose
```

Comment:
⚠️ process.env usage should be memoized
https://expressjs.com/en/advanced/best-practice-performance.html#set-node_env-to-production

```
     name               hz     min     max    mean     p75     p99    p995    p999     rme  samples
   · access   3,449,608.83  0.0003  0.0761  0.0003  0.0003  0.0003  0.0004  0.0005  ±0.21%  1724805
   · cached  14,903,737.94  0.0001  0.0575  0.0001  0.0001  0.0001  0.0001  0.0001  ±0.20%  7451869   fastest
```

Reply:
fixed